src/pudl/analysis/state_demand.py

```diff
@@ -147,7 +147,7 @@ def local_to_utc(local: pd.Series, tz: Iterable, **kwargs: Any) -> pd.Series:
         1   2020-01-01 06:00:00
         dtype: datetime64[ns]
     """
-    return local.groupby(tz).transform(
+    return local.groupby(tz, observed=True).transform(
```

Comment:
Addressing a deprecation warning. There are 15 million records with these timestamps and if we're ever operating on a subset (e.g. a single state that only contains one timezone) this behavior will be much more space efficient.  Also it'll become the default pandas behavior in the future.